### PR TITLE
Make loading of remote list significantly faster

### PR DESCRIPTION
**Pull request by yishaigalatzer**
Optimize away unnecessary network calls, and move them to a background thread.

Based on NuGet/NuGet3#25

### CODE REVIEW

File: src/PackageManagement.UI/Models/SearchResultPackageMetadata.cs [hunk 1]
```diff
@@ -23,14 +25,59 @@ public class SearchResultPackageMetadata : INotifyPropertyChanged
...
         public string Summary { get; set; }
 
+        private bool StatusProviderRun { get; set; }
+
         public PackageStatus Status
         {
-            get { return _status; }
+            get
+            {
+                if (!StatusProviderRun)
```
Comment (yishaigalatzer) [inline src/PackageManagement.UI/Models/SearchResultPackageMetadata.cs:31]: use `StatusProvider.IsValueCreated` instead.

Comment (yishaigalatzer) [inline src/PackageManagement.UI/Models/SearchResultPackageMetadata.cs:31]: After thinking about it, the current code is correct and using IsValueCreated can cause a race condition or missing an update if someone else upstream have touched the Lazy

File: src/PackageManagement.UI/PackageDetailControlModel.cs [hunk 2]
```diff
@@ -22,11 +23,12 @@ public class PackageDetailControlModel : DetailControlModel
             Debug.Assert(nugetProjects.Count() == 1);
         }
 
-        public override void SetCurrentPackage(
+        public async override Task SetCurrentPackage(
```
Comment (deepakaravindr) [inline src/PackageManagement.UI/PackageDetailControlModel.cs:26]: Async suffix. Here and in other places

File: src/PackageManagement.UI/PackageLoader.cs [hunk 3]
```diff
@@ -257,10 +257,10 @@ private IEnumerable<string> GetSupportedFrameworks()
...
         }
 
-        private async Task<UISearchMetadata> GetPackageMetadataAsync(CancellationToken ct,
-                                                               UIMetadataResource localResource,
-                                                               UIMetadataResource metadataResource,
-                                                               PackageIdentity identity)
+        private async Task<UISearchMetadata> GetPackageMetadataAsync(UIMetadataResource localResource,
+                                                                     UIMetadataResource metadataResource,
+                                                                     PackageIdentity identity,
+                                                                     CancellationToken cancellationToen)
```
Comment (deepakaravindr) [inline src/PackageManagement.UI/PackageLoader.cs:263]: typo 'cancellationToken'

**Comment by deepakaravindr** [comment]
:shipit:

**Pull request merged**